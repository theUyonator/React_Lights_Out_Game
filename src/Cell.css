.Cell {
  height: 100px;
  width: 100px;
  background-color: darkgray;
  cursor: pointer;
}

.Cell-lit {
  background-color: #F0CD16;
}