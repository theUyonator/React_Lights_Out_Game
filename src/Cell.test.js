import React from "react";
import { render } from "@testing-library/react";
import Cell from "./Cell";

describe("<Cell /> rendering", () => {
    let container;

beforeEach(() =>{
// We need to add a <tr> element to the cell we are rendring 
// if not react will produce a a warning about appending a
// TD straight to a div.
    let tr = document.createElement("tr");
    container = document.body.appendChild(tr);  

});

it("renders correctly without crashing", () => {
    render(<Cell />, { container })
});

it("matches snapshot when lit", () => {
    const { asFragment } = render(<Cell isLit />, { container });
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when not lit", () => {
    const { asFragment } = render(<Cell />, { container });
    expect(asFragment()).toMatchSnapshot();
});



});