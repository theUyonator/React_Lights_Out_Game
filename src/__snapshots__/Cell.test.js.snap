// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> rendering matches snapshot when lit 1`] = `
<DocumentFragment>
  <td
    class="Cell Cell-lit"
  />
</DocumentFragment>
`;

exports[`<Cell /> rendering matches snapshot when not lit 1`] = `
<DocumentFragment>
  <td
    class="Cell "
  />
</DocumentFragment>
`;
